src/dmd/opover.d

```diff
@@ -853,11 +853,11 @@ Expression op_overload(Expression e, Scope* sc, EXP* pop = null)
                         {
                             // Rewrite (e1 op e2) as e2.opfunc(e1)
                             result = build_overload(e.loc, sc, e.e2, e.e1, m.lastf ? m.lastf : s);
+                            // When reversing operands of comparison operators,
+                            // need to reverse the sense of the op
+                            if (pop)
+                                *pop = reverseRelation(e.op);
                         }
-                        // When reversing operands of comparison operators,
```

Comment:
Is this change necessary? Why?

Reply:
I belive it is: you only need to reverse the `op` only when we decide to write the call exp as `e2.opFunc(e1)`.
As it was, the op was always reversed. Imho this was a bug that just wasn’t manifesting